src/psbt.cpp

```diff
@@ -158,18 +150,6 @@ void PSBTInput::Merge(const PSBTInput& input)
     if (final_script_witness.IsNull() && !input.final_script_witness.IsNull()) final_script_witness = input.final_script_witness;
 }
 
-bool PSBTInput::IsSane() const
-{
-    // Cannot have both witness and non-witness utxos
```

Comment:
In commit "psbt: Allow both non_witness_utxo and witness_utxo" (5279d8bc07d601fe6a67ad665fbc7591fe73c7de):

Previously we didn't allow both witness and non-witness utxos. Now we do allow them but we aren't checking if they are consistent. It seems like instead of removing the IsSane checks, it'd be better to keep them and verify 
`non_witness_utxo->vout[prevout_index] == witness_utxo` if both witness_utxo and non_witness_utxo are present.

Reply:
I think we can add checks in a followup.

I'm also uncertain about adding application logic checks that would disallow deserialization. The PSBT could be correctly formatted but contain the wrong UTXOs or scripts. We might still want to allow `decodepsbt` to show the PSBT, but not allow updating or signing it.

Reply:
Agree, the check would be nice but its not necessary for merge